cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -522,29 +526,13 @@ public void stop(Id.Namespace id, Id.Program programId, ProgramType type) throws
   }
 
   private boolean namespaceExists(String namespace) {
-    boolean isValid = false;
-    if (namespace != null && !namespace.isEmpty()) {
-      if (store.getNamespace(Id.Namespace.from(namespace)) != null) {
-        isValid = true;
-      } else {
-        // null namespace found. check if this is default, and try to create it.
-        // This logic could be changed later to figure out a way to create the 'default' namespace at a better
-        // time during the initialization of CDAP.
-        if (Constants.DEFAULT_NAMESPACE.equals(namespace)) {
-          NamespaceMeta existing = store.createNamespace(new NamespaceMeta.Builder()
-                                                           .setId(Constants.DEFAULT_NAMESPACE)
-                                                           .setName(Constants.DEFAULT_NAMESPACE)
-                                                           .setDescription(Constants.DEFAULT_NAMESPACE).build());
-          if (existing != null) {
-            LOG.trace("Default namespace already exists.");
-          } else {
-            LOG.trace("Created default namespace.");
-          }
-          isValid = true;
-        }
-      }
+    boolean exists = false;
+    try {
+      namespaceService.getNamespace(Id.Namespace.from(namespace));
+      exists = true;
```

Comment:
Hmmm, although in this case there would only be a couple of returns, so it should be ok. fixed.